From 93ac12531f7c672ef1fe7689cf8b67ec2372efef Mon Sep 17 00:00:00 2001
From: Tianling Shen <cnsztl@immortalwrt.org>
Date: Wed, 5 Apr 2023 21:08:27 +0800
Subject: [PATCH 3/3] Revert "rockchip: Convert rockpro64-rk3399 to use
 standard boot"

This reverts commit 3891c68ef50eda38d78c95ecd03aed030aa6bb53.

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>
---
 include/configs/rk3399_common.h   | 5 ++++-
 include/configs/rockchip-common.h | 2 --
 2 files changed, 4 insertions(+), 3 deletions(-)

--- a/include/configs/rk3399_common.h
+++ b/include/configs/rk3399_common.h
@@ -47,12 +47,15 @@
 #define ROCKCHIP_DEVICE_SETTINGS
 #endif
 
+#include <config_distro_bootcmd.h>
+#include <environment/distro/sf.h>
 #define CFG_EXTRA_ENV_SETTINGS \
 	ENV_MEM_LAYOUT_SETTINGS \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"partitions=" PARTS_DEFAULT \
 	ROCKCHIP_DEVICE_SETTINGS \
-	"boot_targets=" BOOT_TARGETS "\0" \
+	BOOTENV \
+	BOOTENV_SF \
 	"altbootcmd=" \
 		"setenv boot_syslinux_conf extlinux/extlinux-rollback.conf;" \
 		"run distro_bootcmd\0"
--- a/include/configs/rockchip-common.h
+++ b/include/configs/rockchip-common.h
@@ -67,14 +67,12 @@
 	BOOT_TARGET_PXE(func) \
 	BOOT_TARGET_DHCP(func) \
 	BOOT_TARGET_SF(func)
-#define BOOT_TARGETS	"mmc1 mmc0 nvme scsi usb pxe dhcp spi"
 #else
 #define BOOT_TARGET_DEVICES(func) \
 	BOOT_TARGET_MMC(func) \
 	BOOT_TARGET_USB(func) \
 	BOOT_TARGET_PXE(func) \
 	BOOT_TARGET_DHCP(func)
-#define BOOT_TARGETS	"mmc1 mmc0 usb pxe dhcp"
 #endif
 
 #ifdef CONFIG_ARM64
